lra/coordinator/server/src/main/java/io/helidon/lra/coordinator/Participant.java

```diff
@@ -0,0 +1,605 @@
+        FAILED_TO_COMPLETE(FailedToComplete, null, null, true, Set.of()),
+
+        CLIENT_COMPENSATING(Compensating, COMPENSATED, FAILED_TO_COMPENSATE, false,
+                Set.of(Active, Compensated, FailedToCompensate)),
+        CLIENT_COMPLETING(Completing, COMPLETED, FAILED_TO_COMPLETE, false, Set.of(Active, Completed, FailedToComplete)),
+        COMPENSATING(Compensating, COMPENSATED, FAILED_TO_COMPENSATE, false, Set.of(Active, Compensated, FailedToCompensate)),
+        COMPLETING(Completing, COMPLETED, FAILED_TO_COMPLETE, false, Set.of(Active, Completed, FailedToComplete));
+
+        private final ParticipantStatus participantStatus;
+        private final Status successFinalStatus;
+        private final Status failedFinalStatus;
+        private final boolean finalState;
+        private final Set<ParticipantStatus> validNextStates;
+
+        Status(ParticipantStatus participantStatus,
+               Status successFinalStatus,
+               Status failedFinalStatus,
+               boolean finalState,
+               Set<ParticipantStatus> validNextStates) {
+            this.participantStatus = participantStatus;
+            this.successFinalStatus = successFinalStatus;
+            this.failedFinalStatus = failedFinalStatus;
+            this.finalState = finalState;
+            this.validNextStates = validNextStates;
+        }
+
+        public ParticipantStatus participantStatus() {
+            return participantStatus;
+        }
+
+        public boolean isFinal() {
+            return finalState;
+        }
+
+        public boolean validateNextStatus(ParticipantStatus participantStatus) {
+            return validNextStates.contains(participantStatus);
+        }
+
+        public Optional<ParticipantStatus> successFinalStatus() {
+            return Optional.ofNullable(successFinalStatus.participantStatus());
+        }
+
+        public Optional<ParticipantStatus> failedFinalStatus() {
+            return Optional.ofNullable(failedFinalStatus.participantStatus);
+        }
+    }
+
+    enum SendingStatus {
+        SENDING, NOT_SENDING;
+    }
+
+    enum AfterLraStatus {
+        NOT_SENT, SENDING, SENT;
+    }
+
+    enum ForgetStatus {
+        NOT_SENT, SENDING, SENT;
+    }
+
+    enum CompensateStatus {
+        NOT_SENT, SENDING, SENT;
+    }
+
+    Participant(Config config) {
+        timeout = config.get("timeout")
+                .asLong()
+                .orElse(500L);
+    }
+
+    void parseCompensatorLinks(String compensatorLinks) {
+        Stream.of(compensatorLinks.split(","))
+                .filter(s -> !s.isBlank())
+                .map(Link::valueOf)
+                .forEach(link -> this.compensatorLinks.put(link.rel(), link.uri()));
+    }
+
+    Optional<URI> getCompensatorLink(String rel) {
+        return Optional.ofNullable(compensatorLinks.get(rel));
+    }
+
+    /**
+     * Invoked when closed 200, 202, 409, 410.
+     */
+    public Optional<URI> getCompleteURI() {
```

Comment:
public method in a package class (see other methods as well)

Reply:
fixed